lighthouse-core/lib/traces/tracing-processor.js

```diff
@@ -246,6 +236,29 @@ class TraceProcessor {
   }
 
   /**
+   * Provides the top level events on the main thread.
+   * @param {!traceviewer.Model} model
+   * @param {{traceEvents: !Array<!Object>}} trace
+   * @param {number=} startTime Optional start time (in ms) of range of interest. Defaults to trace start.
+   * @param {number=} endTime Optional end time (in ms) of range of interest. Defaults to trace end.
+   * @return {!Array<{start: number, end: number, duration: number}>}
+   */
+  static getMainThreadTopLevelEvents(model, trace, startTime = -Infinity, endTime = Infinity) {
+    // Find the main thread via the first TracingStartedInPage event in the trace
+    const startEvent = trace.traceEvents.find(event => event.name === 'TracingStartedInPage');
+    const mainThread = TraceProcessor._findMainThreadFromIds(model, startEvent.pid, startEvent.tid);
+
+    const events = mainThread.sliceGroup.slices.filter(slice => {
+      return slice.title === SCHEDULABLE_TASK_TITLE &&
+          slice.end > startTime &&
+          slice.start < endTime;
+    });
+
+    events.sort((a, b) => a.start - b.start);
```

Comment:
1. i would prefer to do the sort as early as possible as we dont want audits to have to know things are unsorted
1. i'm decently confident that the sort is done by tracingmodel already, so TBH this line can probably just be nuked.